main/src/library/MutDeque.flix

```diff
@@ -87,6 +87,29 @@ namespace MutDeque {
         let MutDeque(_, f, b) = d;
         deref f == deref b
 
+    ///
+    /// Returns the sum of all elements in the deque `d`.
+    ///
+    pub def sum(d: MutDeque[Int32]): Int32 & Impure =
+        sumWith(identity, d)
+
+    ///
+    /// Returns the sum of all elements in the deque `d` according to the function `f`.
+    ///
+    pub def sumWith(f: a -> Int32 & ef, d: MutDeque[a]): Int32 & Impure =
+        // TODO: Refactor when fold is implemented
+        let MutDeque(ar_, front_, back_) = d;
+        let ar = deref ar_;
+        let front = deref front_;
+        let back = deref back_;
+        let c = capacity(d) - 1;
+        def loop(i, e, acc) =
+            if (i == e)
+                acc
+            else
+                loop((i + 1) &&& c, e, f(ar[i]) + acc);
```

Comment:
@jaschdoc What in blazes is going on here with the `&&&`?

Reply:
Haha yeah sorry. It exploits the fact the the size of the backing array is always a power of two. I've been waiting for someone to comment on it haha. It should definitely be changed